### PR TITLE
Feature/raspbian pull

### PR DESCRIPTION
**Pull request by nyuuyn**
#### Short Description
Context: Currently all Implementation Artifacts (IAs) are based on pushing commands to the resources. E.g. SSH connections to VM's and Pis.
One big issue with pushing are firewalls blocking e.g. the SSH Port. To cope with this issue we need to enable IAs that enable pulling such commands instead of pushing. 
E.g. a MQTT Broker that is used as an IA may be used to publish commands on topics, and afterwards, are pulled and executed by the respective resources.

This pull request adds features to the container to enable such IAs. The following features are implemented right now:

Enables Output Parameter handling of Script Artifacts. (Management Bus Change)
Adds configuration of a global hostname to container runtime (enables IAs to expose topics accessible over the net => enables resources to pull commands from anywhere as long they have internet)

### CODE REVIEW

File: docker/config.ini.tpl [hunk 1]
```diff
@@ -11,6 +11,7 @@ eclipse.p2.profile=DefaultProfile
 eclipse.p2.data.area=@config.dir/../p2
 equinox.use.ds=true
 org.eclipse.equinox.simpleconfigurator.configUrl=file\:org.eclipse.equinox.simpleconfigurator/bundles.info
+org.opentosca.container.engine.ia.publicip={{ .Env.PUBLIC_IP }}
```
Comment (miwurster) [inline docker/config.ini.tpl]: Why do we need an additional environment variable? We already have `ENGINE_IA_HOSTNAME` which is basically intended to be the public hostname/IP address of the IA engine.

Btw, I would have called it `org.opentosca.container.engine.ia.hostname`...

File: org.opentosca.bus.management.api.soaphttp/src/org/opentosca/bus/management/api/soaphttp/processor/RequestProcessor.java [hunk 2]
```diff
@@ -83,6 +83,9 @@ public void process(final Exchange exchange) throws Exception {
             }
         }
 
+        // add header with public IP of engine-ia for the raspbian pull IA
+        exchange.getIn().setHeader("OPENTOSCA_PUBLIC_IP", Settings.OPENTOSCA_PUBLIC_IP);
```
Comment (miwurster) [inline org.opentosca.bus.management.api.soaphttp/src/org/opentosca/bus/management/api/soaphttp/processor/RequestProcessor.java]: Follow-up from above; AFAIK this is the public hostname of the IA engine, right? So, we also should call the constant accordingly.

File: org.opentosca.bus.management.plugins.remote.service.impl/src/org/opentosca/bus/management/plugins/remote/service/impl/ManagementBusPluginRemoteServiceImpl.java [hunk 3]
```diff
@@ -649,7 +649,14 @@ private String createParamsString(final Object params, final String publicIP) {
                 final String replacedValue = param.getValue().replaceFirst("container", publicIP);
```
Comment (miwurster) [inline org.opentosca.bus.management.plugins.remote.service.impl/src/org/opentosca/bus/management/plugins/remote/service/impl/ManagementBusPluginRemoteServiceImpl.java]: Strange that we need this adaption at all. If you configure and start the container correctly this shouldn't be necessary, right?

File: docker/config.ini.tpl (hunk 1 above)
Comment (nyuuyn) [reply docker/config.ini.tpl]: So we should just set this to a public IP? There will be no side-effects between the docker containers e.g. engine-ia:8080 whatever?

**Pull request merged**